### PR TITLE
S3 deploy

### PR DESCRIPTION
**Pull request by maxdavidson**
http://paintbot-webapp-20181108155439999600000001.s3-website-eu-west-1.amazonaws.com/

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -95,9 +122,13 @@ workflows:
...
             - install
-      - deploy:
+      - approve-deploy:
+          type: approval
           requires:
             - typecheck
             - lint
             - test
             - build
+      - deploy:
```
Comment (palmithor) [inline .circleci/config.yml]: Might be smart to only allow this to be done on certain branches?

File: infrastructure/main.tf [hunk 2]
```diff
@@ -0,0 +1,43 @@
+provider "aws" {
+  version = "~> 1.43"
+  region  = "${var.aws_region}"
+}
+
+terraform {
+  backend "s3" {
```
Comment (palmithor) [inline infrastructure/main.tf]: 👏

File: .circleci/config.yml (hunk 1 above)
Comment (maxdavidson) [reply .circleci/config.yml]: Sure, but I wanted to test the deploy first 😅 We only have one environment right now, so deploying on both `develop` and `master` is fine.

File: .circleci/config.yml [hunk 3]
```diff
@@ -64,19 +64,46 @@ jobs:
...
+            unzip terraform_0.11.10_linux_amd64.zip
+      - run:
+          name: Initialize Terraform
+          command: ./terraform init
+      - run:
+          name: Create Terraform plan
+          command: ./terraform plan --out terraform.plan
+      - run:
+          name: Apply Terraform plan
+          command: ./terraform apply --auto-approve terraform.plan
```
Comment (maxdavidson) [inline .circleci/config.yml]: CircleCI only allows approval steps between jobs, so I would have to pull these steps apart and persist the terraform executable. I would also like to inject Terraform's `website_endpoint` output into `PUBLIC_URL` in the build step, but that would be even messier. 🤔

**Pull request merged**